### PR TITLE
{bio}[foss/2018b] RSeQC v3.0.0, bx-python v0.8.2, pyBigWig v0.3.13 w/ Python 3.6.6

### PR DESCRIPTION
**Pull request by SethosII**
(created using `eb --new-pr`)

### CODE REVIEW

File: easybuild/easyconfigs/b/bx-python/bx-python-0.8.2-foss-2018b-Python-3.6.6.eb [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+    }),
+    (name, version, {
+        'modulename': 'bx',
+        'source_urls': ['https://pypi.python.org/packages/source/b/%(name)s/'],
+        'checksums': ['faeb0c7c9fcb2f95c4fc1995af4f45287641deee43a01659bd30fe95c5d37386'],
+    }),
+]
+
+# specify that Bundle easyblock should run a full sanity check, rather than just trying to load the module
+full_sanity_check = True
```
Comment (vanzod) [inline easybuild/easyconfigs/b/bx-python/bx-python-0.8.2-foss-2018b-Python-3.6.6.eb]: Missing
```
use_pip = True
```

File: easybuild/easyconfigs/b/bx-python/bx-python-0.8.2-foss-2018b-Python-3.6.6.eb [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+    ('Python', '3.6.6'),
+    ('LZO', '2.10'),
+]
+
+# this is a bundle of Python packages
+exts_defaultclass = 'PythonPackage'
+
+prebuildopts = "export CPATH=$EBROOTLZO/include/lzo:$CPATH && "
+
+exts_download_dep_fail = True
```
Comment (vanzod) [inline easybuild/easyconfigs/b/bx-python/bx-python-0.8.2-foss-2018b-Python-3.6.6.eb]: Missing
```
use_pip = True
```

File: easybuild/easyconfigs/b/bx-python/bx-python-0.8.2-foss-2018b-Python-3.6.6.eb [hunk 3]
```diff
@@ -0,0 +1,48 @@
+easyblock = 'Bundle'
```
Comment (boegel) [inline easybuild/easyconfigs/b/bx-python/bx-python-0.8.2-foss-2018b-Python-3.6.6.eb]: @SethosII Please consider changing this to `PythonBundle` (& clean up the easyconfig accordingly, see #7124)

File: easybuild/easyconfigs/p/pyBigWig/pyBigWig-0.3.13-foss-2018b-Python-3.6.6.eb [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+download_dep_fail = True
+
+sanity_check_paths = {
+    'files': [],
+    'dirs': ['lib/python%(pyshortver)s/site-packages'],
+}
+
+sanity_check_commands = ['python lib/python%(pyshortver)s/site-packages/pyBigWigTest/test.py']
+
+modextrapaths = {'PYTHONPATH': ['lib/python%(pyshortver)s/site-packages']}
```
Comment (boegel) [inline easybuild/easyconfigs/p/pyBigWig/pyBigWig-0.3.13-foss-2018b-Python-3.6.6.eb]: @SethosII This line is not needed, it's taken care of automatically with `PythonPackage`

File: easybuild/easyconfigs/r/RSeQC/RSeQC-3.0.0-foss-2018b-Python-3.6.6.eb [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+    ('pyBigWig', '0.3.13', versionsuffix),
+]
+
+use_pip = True
+
+download_dep_fail = True
+
+sanity_check_paths = {
+    'files': ['bin/bam_stat.py', 'bin/overlay_bigwig.py', 'bin/split_paired_bam.py'],
+    'dirs': ['lib']
```
Comment (boegel) [inline easybuild/easyconfigs/r/RSeQC/RSeQC-3.0.0-foss-2018b-Python-3.6.6.eb]: should be `'lib/python%(pyshortver)s/site-packages'`?

**Comment by SethosII** [comment]
@boegel @vanzod I incoporated your remarks.

**Comment by vanzod** [comment]
Test report by @vanzod
**FAILED**
Build succeeded for 1 out of 3 (3 easyconfigs in this PR)
cermis - Linux debian 9.7, Intel(R) Xeon(R) CPU           X5670  @ 2.93GHz, Python 2.7.13
See https://gist.github.com/c3aed9024aaf5497fc7d21a0ae882609 for a full test report.

**Comment by vanzod** [comment]
@SethosII It looks like `pyBigWig` requires `cURL` as dependency

**Comment by SethosII** [comment]
@vanzod I added `cURL` as a dependency.

**Comment by vanzod** [comment]
Test report by @vanzod
**SUCCESS**
Build succeeded for 3 out of 3 (3 easyconfigs in this PR)
cermis - Linux debian 9.7, Intel(R) Xeon(R) CPU           X5670  @ 2.93GHz, Python 2.7.13
See https://gist.github.com/b42219584b6c89e5c931c76920b60c96 for a full test report.

File: easybuild/easyconfigs/b/bx-python/bx-python-0.8.2-foss-2018b-Python-3.6.6.eb [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+        'modulename': 'bx',
+        'source_urls': ['https://pypi.python.org/packages/source/b/%(name)s/'],
+        'checksums': ['faeb0c7c9fcb2f95c4fc1995af4f45287641deee43a01659bd30fe95c5d37386'],
+    }),
+]
+
+sanity_check_paths = {
+    'files': [],
+    'dirs': ['bin', 'lib/python%(pyshortver)s/site-packages/bx_python-%(version)s-py%(pyshortver)s.egg-info',
+             'lib/python%(pyshortver)s/site-packages/python_lzo-1.12-py%(pyshortver)s.egg-info'],
```
Comment (boegel) [inline easybuild/easyconfigs/b/bx-python/bx-python-0.8.2-foss-2018b-Python-3.6.6.eb]: @SethosII We usually only check for `'lib/python%(pyshortver)s/site-packages'` here, more specific checks are prone to get broken (in particular when versions of required dependencies are hardcoded like you did here for `python-lzo`)

Comment (SethosII) [reply easybuild/easyconfigs/b/bx-python/bx-python-0.8.2-foss-2018b-Python-3.6.6.eb]: @boegel The intention was to check that both packages of the bundle are there. Is there a way to use the `exts_list` variable for that? I removed it for now.

Comment (boegel) [reply easybuild/easyconfigs/b/bx-python/bx-python-0.8.2-foss-2018b-Python-3.6.6.eb]: EasyBuild already does an `import` check for each extension. There have been cases where Python packages get installed slightly differently under some circumstances (e.g. as a zipped `.egg` *file* vs an (unzipped) `.egg` directory), which is why we've opted for only checking the parent directory.

In theory the `import` could still work if an extension accidentally was installed somewhere else instead, but we can look into enhancing the `import` check to catch those too (by checking where the module was imported from).

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 3 out of 3 (3 easyconfigs in this PR)
node3118.skitty.os - Linux centos linux 7.6.1810, Intel(R) Xeon(R) Gold 6140 CPU @ 2.30GHz, Python 2.7.5
See https://gist.github.com/2831af2802ff88ed710dcfa84be0c79b for a full test report.

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 3 out of 3 (3 easyconfigs in this PR)
node2431.golett.os - Linux centos linux 7.6.1810, Intel(R) Xeon(R) CPU E5-2680 v3 @ 2.50GHz, Python 2.7.5
See https://gist.github.com/b56524daf0c01915c6120e4f1e383571 for a full test report.

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 3 out of 3 (3 easyconfigs in this PR)
node2016.delcatty.os - Linux centos linux 7.6.1810, Intel(R) Xeon(R) CPU E5-2670 0 @ 2.60GHz, Python 2.7.5
See https://gist.github.com/4771237b4bf0ccc4841d4a59281e776a for a full test report.

**Comment by boegel** [comment]
Going in, thanks @SethosII!

**Pull request merged**